tests/components/dexcom/test_config_flow.py

```diff
@@ -0,0 +1,130 @@
+"""Test the Dexcom config flow."""
+from pydexcom import AccountError, SessionError
+
+from homeassistant import config_entries, data_entry_flow, setup
+from homeassistant.components.dexcom.const import DOMAIN, MG_DL, MMOL_L
+from homeassistant.const import CONF_UNIT_OF_MEASUREMENT, CONF_USERNAME
+
+from tests.async_mock import patch
+from tests.common import MockConfigEntry
+from tests.components.dexcom import CONFIG
+
+
+async def test_form(hass):
+    """Test we get the form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["errors"] == {}
+
+    with patch(
+        "homeassistant.components.dexcom.config_flow.Dexcom.create_session",
+        return_value="test_session_id",
+    ), patch(
+        "homeassistant.components.dexcom.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.dexcom.async_setup_entry", return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], CONFIG,
+        )
+
+    assert result2["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5760157)